### PR TITLE
share_util: Remove usages of deprecated API

### PR DESCRIPTION
**Pull request by alexander-polovtcev**

### CODE REVIEW

File: genestack_client/share_util.py [hunk 1]
```diff
@@ -76,7 +78,9 @@ def share_files_for_edit(self, file_accessions, group_accession, destination_fol
                will be created if this parameter is equal to `None`.
         :type destination_folder: str
         """
-        self.__share(file_accessions, group_accession, destination_folder, 'shareFilesForEditing')
+        self.safe_share_files(
+            file_accessions, group_accession, [ShareUtil.Permissions.EDIT], destination_folder
```
Comment (genestack-solomatin) [inline genestack_client/share_util.py]: Do we need to add view permission here?

**Comment by alexander-polovtcev** [comment]
> Do you plan to

Please, continue

File: genestack_client/share_util.py (hunk 1 above)
Comment (alexander-polovtcev) [reply genestack_client/share_util.py]: No, these permissions are not the same as in the core, so EDIT includes VIEW as well

**Comment by genestack-solomatin** [comment]
> Please, continue

No.

**Pull request merged**